File: 2632.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
var prompt = function(texto) { return lines.shift();};

const table = {
  fire: {
    dmg: 200,
    1: 20,
    2: 30,
    3: 50,
  },
  water: {
    dmg: 300,
    1: 10,
    2: 25,
    3: 40,
  },
  earth: {
    dmg: 400,
    1: 25,
    2: 55,
    3: 70,
  },
  air: {
    dmg: 100,
    1: 18,
    2: 38,
    3: 60,
  },
};

var cases = parseInt(prompt());



for (let i = 0; i < cases; i++) {
  var [width, height, minX, minY] = prompt().split(" ").map(Number);
  var [spell, lvl, centerX, centerY] = prompt().split(" ");
  [centerX, centerY] = [centerX, centerY].map(Number);

  var [maxX, maxY] = [minX + width, minY + height];

  var distanceX = Math.max(minX - centerX, 0, centerX - maxX);
  var distanceY = Math.max(minY - centerY, 0, centerY - maxY);
  var totalDistance = Math.sqrt(distanceX * distanceX + distanceY * distanceY);

  if (totalDistance <= table[spell][lvl]) {
    console.log(table[spell].dmg);
  } else {
    console.log(0);
  }
}

File: java/2632.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out = new PrintWriter(System.out);
        
        int cases = Integer.parseInt(br.readLine().trim());
        
        for (int i = 0; i < cases; i++) {
            String[] dimensions = br.readLine().trim().split(" ");
            int width = Integer.parseInt(dimensions[0]);
            int height = Integer.parseInt(dimensions[1]);
            int minX = Integer.parseInt(dimensions[2]);
            int minY = Integer.parseInt(dimensions[3]);
            
            String[] spellInfo = br.readLine().trim().split(" ");
            String spell = spellInfo[0];
            int level = Integer.parseInt(spellInfo[1]);
            int centerX = Integer.parseInt(spellInfo[2]);
            int centerY = Integer.parseInt(spellInfo[3]);
            
            int maxX = minX + width;
            int maxY = minY + height;
            
            
            int distanceX = Math.max(minX - centerX, Math.max(0, centerX - maxX));
            int distanceY = Math.max(minY - centerY, Math.max(0, centerY - maxY));
            double totalDistance = Math.sqrt(distanceX * distanceX + distanceY * distanceY);
            
         
            int range = 0;
            int damage = 0;
            
            switch (spell) {
                case "fire":
                    damage = 200;
                    if (level == 1) range = 20;
                    else if (level == 2) range = 30;
                    else if (level == 3) range = 50;
                    break;
                case "water":
                    damage = 300;
                    if (level == 1) range = 10;
                    else if (level == 2) range = 25;
                    else if (level == 3) range = 40;
                    break;
                case "earth":
                    damage = 400;
                    if (level == 1) range = 25;
                    else if (level == 2) range = 55;
                    else if (level == 3) range = 70;
                    break;
                case "air":
                    damage = 100;
                    if (level == 1) range = 18;
                    else if (level == 2) range = 38;
                    else if (level == 3) range = 60;
                    break;
            }
            
            if (totalDistance <= range) {
                out.println(damage);
            } else {
                out.println(0);
            }
        }
        
        out.close();
        br.close();
    }
}
